src/plugins/intel_gpu/src/graph/strided_slice.cpp

```diff
@@ -44,8 +44,20 @@ std::vector<layout> strided_slice_inst::calc_output_layouts(strided_slice_node c
     if ((begin_data.empty() && !constant_mem.count(1))
         || (end_data.empty() && !constant_mem.count(2))
         || (strides_data.empty() && !constant_mem.count(3))) {
-        auto out_shape = ov::PartialShape::dynamic(input0_layout.get_partial_shape().size());
-        return { layout{out_shape, input0_layout.data_type, format::get_default_format(out_shape.rank().get_length())} };
+        auto input0_pshape = input0_layout.get_partial_shape();
+        auto input0_len = input0_pshape.size();
+        auto out_shape = ov::PartialShape::dynamic(input0_len);
+        if (input0_layout.is_dynamic()) {
```

Comment:
I think its always dynamic, isnt it?

Reply:
input0 could be static while input1/input2/input3 is dynamic

Reply:
If input0 could be static, how to set out_shape? in this code, out_shape[] is not filled when input0_layout is static.
it only fills out_shape[] when input0_layout is dynamic.

Reply:
Yeah at first I misread it as output. BTW I also would like to ask how static input could be handled as Paul mentioned? I think input's dynamism does not matter.